### PR TITLE
docs(ComponentDoc): refactor and optimize

### PR DESCRIPTION
**Pull request by layershifter**
_Splitted from #2070 for the easier review of proposed changes._

This PR is part of work (#1981, #2012) that improves docs usability and performance. Contains the deep refactor of `ComponentDoc`.

### Naming

This PR includes refactor of `ComponentDoc`. As before, I've refactored it to smaller components and move down `showPropsFor` to `ComponentProps`. But I met naming problem and I hope for your help there.

- `ComponentProps` => `ComponentTable` - after performed refactoring, this component has been renamed, since by context it is a **table** with of props
- `ComponentProps/*` => `ComponentProp/*` - there are numerous small components that represent props, I moved them there and it's what makes me doubt

### CODE REVIEW

File: gulp/plugins/gulp-react-docgen.js [hunk 1]
```diff
@@ -52,9 +52,13 @@ export default (filename) => {
...
         }
       })
+
+      parsed.path = file.path
+        .replace(`${process.cwd()}/`, '')
+        .replace(new RegExp(_.escapeRegExp(path.sep), 'g'), '/')
       parsed.props = _.sortBy(parsed.props, 'name')
 
-      result[relativePath] = parsed
+      result[componentName] = parsed
```
Comment (layershifter) [inline gulp/plugins/gulp-react-docgen.js]: `posixPath` is now calculated by plugin. For `key` in `docgenInfo.json` will be used a component name, it's unique, so we don't to regexps to find the component info :v:

**Comment by layershifter** [comment]
@levithomason It's another try, it's different from that you see in #2070.

### `withDocInfo`

Now we have the HOC that generates all needed info about component, later we can easily refactor it to support codesplitting :+1:

File: gulp/plugins/gulp-react-docgen.js (hunk 1 above)
Comment (levithomason) [reply gulp/plugins/gulp-react-docgen.js]: We need to make it windows compatible, so it needs to use Noed's `path` module for path modification.  We cannot hardcode forward or backward slashes.

Comment (layershifter) [reply gulp/plugins/gulp-react-docgen.js]: I've shipped a small change, looks correct. I don't have a windows PC near and I'm not familar with `path`. If you have better solution, feel free to push updates.

**Comment by levithomason** [comment]
There are still some windows build breaking forward slashes in here, but we'll address those if we hear from windows users.

**Comment by levithomason** [comment]
Released in `semantic-ui-react@0.76.0`

**Pull request merged**